src/chat-handler.ts

```diff
@@ -9,23 +9,30 @@ import {
   IChatMessage,
   INewMessage
 } from '@jupyter/chat';
-import { UUID } from '@lumino/coreutils';
-import type { ChatMistralAI } from '@langchain/mistralai';
+import type { BaseChatModel } from '@langchain/core/language_models/chat_models';
 import {
   AIMessage,
   HumanMessage,
   mergeMessageRuns
 } from '@langchain/core/messages';
+import { UUID } from '@lumino/coreutils';
 
 export type ConnectionMessage = {
   type: 'connection';
   client_id: string;
 };
 
-export class CodestralHandler extends ChatModel {
-  constructor(options: CodestralHandler.IOptions) {
+export class ChatHandler extends ChatModel {
+  constructor(options: ChatHandler.IOptions) {
     super(options);
-    this._mistralClient = options.mistralClient;
+    this._llmClient = options.llmClient;
+  }
+
+  get llmClient(): BaseChatModel | null {
```

Comment:
Wondering if "provider" could also be a good name as an alternative to `llmClient`? Or something else similar to what Jupyter AI uses?